### PR TITLE
Update foreman_openscap to 0.3.0

### PR DESCRIPTION
**Pull request by isimluk**
I have checked that it builds on my local RHEL6 box. The scratch build in koji is missing though.

### CODE REVIEW

File: rubygem-foreman_openscap/rubygem-foreman_openscap.spec [hunk 1]
```diff
@@ -26,7 +26,7 @@ Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
 Requires: foreman >= 1.5.0
 
 Requires: %{?scl_prefix}rubygem(deface)
-Requires: %{?scl_prefix}rubygem(scaptimony) >= 0.2.0
+Requires: %{?scl_prefix}rubygem(scaptimony) = 0.3.0
```
Comment (domcleal) [inline rubygem-foreman_openscap/rubygem-foreman_openscap.spec]: There shouldn't be a version spec here, the gem requires ~> 0

**Comment by domcleal** [comment]
Could you use git annex add to when committing the gem, instead of committing the binary itself?  There are a few instructions in the README which should help you set it up and show the commands, shout if you get stuck.

**Comment by isimluk** [comment]
Updated new changeset. Thanks for review.

**Comment by domcleal** [comment]
[test]

File: rubygem-foreman_openscap/rubygem-foreman_openscap.spec [hunk 2]
```diff
@@ -39,9 +39,6 @@ BuildRequires: %{?scl_prefix}ruby(abi) >= %{rubyabi}
 BuildRequires: %{?scl_prefix}rubygems-devel
 BuildRequires: %{?scl_prefix}rubygems
 BuildRequires: foreman-assets >= 1.7.0
-# TODO could be removed later when dependencies are fixed https://github.com/OpenSCAP/foreman_openscap/issues/67
```
Comment (domcleal) [inline rubygem-foreman_openscap/rubygem-foreman_openscap.spec]: I don't think this comment was correct actually, they will still be build-deps for asset precompilation.  The tests should confirm!

**Comment by isimluk** [comment]
Yep, you are right. it fails in my copr https://copr.fedoraproject.org/coprs/isimluk/OpenSCAP/build/69599/

**Comment by domcleal** [comment]
Okay, if you could revert those lines then we'll re-test.

**Comment by isimluk** [comment]
Ok, reverted, but patch is still not ready. The assets precompile does not create assets in public/assets/foreman_openscap directory.

```
RPM build errors:
    File not found: /root/rpmbuild/BUILDROOT/ruby193-rubygem-foreman_openscap-0.3.0-1.el6.x86_64/opt/rh/ruby193/root/usr/share/gems/gems/foreman_openscap-0.3.0/public/assets/foreman_openscap
    File not found: /root/rpmbuild/BUILDROOT/ruby193-rubygem-foreman_openscap-0.3.0-1.el6.x86_64/opt/rh/ruby193/root/usr/share/gems/gems/foreman_openscap-0.3.0/public/assets/foreman_openscap
```

Also, I have decided to put SCAPtimony version requirements here. It will not work with other versions anyway. (Fixing in upstream as well OpenSCAP/foreman_openscap#72).

**Comment by domcleal** [comment]
Thanks for updating the version requirements in the gemspec too, it's important they stay in sync otherwise it can cause runtime errors.

It seems to build in mock just fine, so let's see if the PR [test] on Koji works.

**Comment by domcleal** [comment]
Looks fine to me, did you want to investigate more or are you happy for me to merge this?

**Comment by isimluk** [comment]
Please merge as it builds in koji. I will investigate issue with my setup on my own. Thanks!

File: rubygem-foreman_openscap/rubygem-foreman_openscap.spec [hunk 3]
```diff
@@ -105,13 +106,14 @@ mkdir -p %{buildroot}%{foreman_bundlerd_dir}
 
 %posttrans
 # We need to run the db:migrate (because of SCAPtimony) after the install transaction
-/usr/sbin/foreman-rake db:migrate  >/dev/null 2>&1 || :
-/usr/sbin/foreman-rake db:seed  >/dev/null 2>&1 || :
```
Comment (domcleal) [inline rubygem-foreman_openscap/rubygem-foreman_openscap.spec]: You need to keep db:seed as there's a seed file in the plugin

File: rubygem-foreman_openscap/rubygem-foreman_openscap.spec [hunk 4]
```diff
@@ -105,13 +106,14 @@ mkdir -p %{buildroot}%{foreman_bundlerd_dir}
...
-/usr/sbin/foreman-rake apipie:cache  >/dev/null 2>&1 || :
```
Comment (domcleal) [inline rubygem-foreman_openscap/rubygem-foreman_openscap.spec]: ditto apipie:cache as there's an API controller

File: rubygem-foreman_openscap/rubygem-foreman_openscap.spec [hunk 5]
```diff
@@ -105,13 +106,14 @@ mkdir -p %{buildroot}%{foreman_bundlerd_dir}
...
-(/sbin/service foreman status && /sbin/service foreman restart) >/dev/null 2>&1
+%foreman_db_migrate
```
Comment (domcleal) [inline rubygem-foreman_openscap/rubygem-foreman_openscap.spec]: but no DB migrations!

Comment (domcleal) [inline rubygem-foreman_openscap/rubygem-foreman_openscap.spec]: Ack, cancel that.. there are in scaptimony aren't there?  So leave this line in, but the others need to stay, or be replaced by macros if you like.

**Comment by domcleal** [comment]
Thanks @isimluk!  Will build it now.

**Pull request merged**